pkg/summarizer/naive/analyzer_test.go

```diff
@@ -0,0 +1,355 @@
+		expectedTestInfo *summarypb.TestInfo
+		expectedSuccess  bool
+	}{
+		{
+			name:             "correctly filters Result with less than minRuns",
+			test:             &types.Result{},
+			minRuns:          1000, // arbitrarily large number so that it should get filtered
+			expectedTestInfo: &summarypb.TestInfo{},
+			expectedSuccess:  false,
+		},
+		{
+			name: "typical Result returns correct TestInfo",
+			test: &types.Result{
+				Passed:           3,
+				Failed:           2,
+				FlakyCount:       8,
+				AverageFlakiness: 50.0,
+				FailedInfraCount: 4,
+				InfraFailures: map[string]int{
+					"infra1": 3,
+					"infra2": 1,
+				},
+			},
+			minRuns: -1,
+			expectedTestInfo: &summarypb.TestInfo{
+				Name:               "",
+				Env:                "",
+				Flakiness:          40.0,
+				TotalRuns:          5,
+				TotalRunsWithInfra: 9,
+				PassedRuns:         3,
+				FailedRuns:         2,
+				FailedInfraRuns:    4,
+				InfraIssues:        "infra1 75.00% infra2 25.00%",
+			},
+			expectedSuccess: true,
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			if actualTest, actualSuccess := calculateNaiveFlakiness(tc.test, tc.minRuns); !proto.Equal(actualTest, tc.expectedTestInfo) || tc.expectedSuccess != actualSuccess {
+				t.Errorf("\ntestInfo:\nactual: %v vs. expected: %v\nsuccess:\nactual: %v vs. expected: %v", actualTest, tc.expectedTestInfo, actualSuccess, tc.expectedSuccess)
+			}
+		})
+	}
+}
+
+func TestCalculateInfraInfo(t *testing.T) {
+	cases := []struct {
+		name        string
+		issues      map[string]int
+		failedCount int
+		expected    string
+	}{
+		{
+			name:        "empty issues map returns empty string",
+			issues:      map[string]int{},
+			failedCount: 10,
+			expected:    "",
+		},
+		{
+			name: "zero failedCount returns empty string",
+			issues: map[string]int{
+				"infra1": 2,
+				"infra2": 7,
+			},
+			failedCount: 0,
+			expected:    "",
+		},
+		{
+			name: "typical issues map returns properly formatted string",
+			issues: map[string]int{
+				"infra1": 2,
+				"infra2": 7,
+			},
+			failedCount: 10,
+			expected:    "infra2 70.00% infra1 20.00%",
+		},
+		{
+			name: "issues with same value count orders on key, ascending",
+			issues: map[string]int{
+				"infra1": 2,
+				"infra2": 2,
+			},
+			failedCount: 10,
+			expected:    "infra1 20.00% infra2 20.00%",
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			if actual := calculateInfraInfo(tc.issues, tc.failedCount); actual != tc.expected {
+				t.Errorf("actual: %s != expected: %s", actual, tc.expected)
+			}
+		})
+	}
+}
+
+func TestIntToTimestamp(t *testing.T) {
+	cases := []struct {
+		name     string
+		seconds  int
+		expected *timestamp.Timestamp
+	}{
+		{
+			name:    "typical input returns correct timestamp",
+			seconds: 2,
+			expected: &timestamp.Timestamp{
+				Seconds: int64(2),
+			},
+		},
+		{
+			name:    "typical input returns correct timestamp, test 2",
```

Comment:
(ditto - in general we don't need to test the same functionality twice)

Reply:
Done